plugin/src/cli_utils/localpvzfs/volume/mod.rs

```diff
@@ -0,0 +1,118 @@
+use types::{ZfsVolRecord, ZfsVolume, ZfsVolumeObject};
+
+use k8s_openapi::api::core::v1::PersistentVolume;
+use kube::{api::ListParams, Api, Client};
+use prettytable::{row, Row};
+
+use lazy_static::lazy_static;
+
+lazy_static! {
+    pub(crate) static ref ZFS_VOLUME_HEADER: Row =
+        row!["NAME", "NODE", "STATUS", "CAPACITY", "POOL", "PVC-NAME", "SC-NAME",];
+}
+
+/// Implementation for volumes cmd.
+pub(crate) async fn volumes(cli_args: &CliArgs, volumes_arg: &GetVolumesArg) -> Result<(), Error> {
+    let client = Client::try_default()
+        .await
+        .map_err(|err| Error::Kube { source: err })?;
+    let volume_handle: Api<ZfsVolume> = Api::namespaced(client.clone(), &cli_args.args.namespace);
+    let vols = zfs_volumes(volume_handle, volumes_arg)
+        .await
+        .map_err(|err| Error::Kube { source: err })?;
+    match get_zfs_vol_record(vols, &client).await {
+        Ok(zvol_record) => {
+            print_table(&cli_args.output, zvol_record);
+            Ok(())
+        }
+        Err(e) => Err(e),
+    }
+}
+
+/// Implementation for volume cmd.
+pub(crate) async fn volume(cli_args: &CliArgs, volume_arg: &GetVolumeArg) -> Result<(), Error> {
+    let client = Client::try_default()
+        .await
+        .map_err(|err| Error::Kube { source: err })?;
+    let volume_handle: Api<ZfsVolume> = Api::namespaced(client.clone(), &cli_args.args.namespace);
+    let volume = zfs_volume(volume_handle, volume_arg)
+        .await
+        .map_err(|err| Error::Kube { source: err })?;
+    match get_zfs_vol_record(vec![volume], &client).await {
+        Ok(zvol_record) => {
+            print_table(&cli_args.output, zvol_record);
+            Ok(())
+        }
+        Err(e) => Err(e),
+    }
+}
+
+/// Gets a specific zfs volume fom k8s.
+async fn zfs_volume(
+    volume_handle: Api<ZfsVolume>,
+    volume_arg: &GetVolumeArg,
+) -> Result<ZfsVolume, kube::Error> {
+    let volume = volume_handle.get(&volume_arg.volume).await?;
+    Ok(volume)
+}
+
+/// Lists zfsvolume cr. Filters based on node if specified.
+async fn zfs_volumes(
+    volume_handle: Api<ZfsVolume>,
+    volumes_arg: &GetVolumesArg,
+) -> Result<Vec<ZfsVolume>, kube::Error> {
+    let lp = if let Some(node_id) = volumes_arg.node_id.clone() {
+        ListParams::default().labels(format!("kubernetes.io/nodename={}", node_id).as_str())
+    } else {
+        ListParams::default()
+    };
+    Ok(volume_handle.list(&lp).await?.items)
+}
+
+/// Converts Vec<ZfsVolume> into ZfsRecord.
+async fn get_zfs_vol_record(
+    zfs_vols: Vec<ZfsVolume>,
+    client: &Client,
+) -> Result<ZfsVolRecord, Error> {
+    let api: Api<PersistentVolume> = Api::<PersistentVolume>::all(client.clone());
+    let mut zfs_volumes: Vec<ZfsVolumeObject> = Vec::new();
+    for zfs_vol in zfs_vols {
+        let pv = api
+            .get(zfs_vol.name_any().as_str())
+            .await
+            .map_err(|err| Error::Kube { source: err })?;
```

Comment:
Yeah, Csi controller creates lvm/zfs volume cr. If backend resource cannot be created it will delete the cr. 

In this workflow, PV is not created but cr will come and go depending on /CreateVolume backoff.

We will just print error on terminal for CRs whose PV is not found and move on.